<?xml version="1.0"?>
<launch>
	<!--lcl-->
	<param name="lcl/save_num" type="int" value="50" />
	<node pkg="sensor_fusion" type="lcl" name="lcl" >
		<remap from="/cloud" to="/cloud/tf" />
	</node>
    
    <!--Divide PointCloud-->
    <node pkg="sensor_fusion" type="division" name="division">
        <remap from="/cloud"        to="/cloud/lcl" />
        <remap from="/cloud/center" to="/sq_lidar/points/center" />
        <remap from="/cloud/left"   to="/sq_lidar/points/left" />
        <remap from="/cloud/right"  to="/sq_lidar/points/right" />
    </node>

	<!--Transform PointCloud-->
	<node pkg="sensor_fusion" type="laser_transform_pointcloud" name="transform_pointcloud_center">
		<param name="target_frame" type="string" value="zed0/zed_left_camera" />
        <remap from="/cloud"    to="/sq_lidar/points/center" />
		<remap from="/cloud/tf" to="/sq_lidar/points/center/tf" />
	</node>

	<!--Transform PointCloud-->
	<node pkg="sensor_fusion" type="laser_transform_pointcloud" name="transform_pointcloud_right">
		<param name="target_frame" type="string" value="zed1/zed_left_camera" />
        <remap from="/cloud"    to="/sq_lidar/points/right" />
		<remap from="/cloud/tf" to="/sq_lidar/points/right/tf" />
	</node>

	<!--Transform PointCloud-->
	<node pkg="sensor_fusion" type="laser_transform_pointcloud" name="transform_pointcloud_left">
        <param name="target_frame" type="string" value="zed2/zed_left_camera" />
        <remap from="/cloud"    to="/sq_lidar/points/left" />
		<remap from="/cloud/tf" to="/sq_lidar/points/left/tf" />
	</node>


</launch>
